helpers/setURLQueryParam.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+const URL = require('url').URL;
+const utils = require('handlebars-utils');
+const common = require('./lib/common.js');
+
+const factory = () => {
+    return function(string, key, value) {
+        if (!utils.isString(string) || !common.isValidURL(string)
+            || !utils.isString(key) || !utils.isString(value)
+            || !(key.length > 0) || !(value.length > 0)) {
+            // Return empty string if this does not appear to be a URL or if the
+            // key/value are not strings of at least 1 character in length
+            return ''
+        }
+
+        const url = new URL(string);
+
+        url.searchParams.set(encodeURIComponent(key), encodeURIComponent(value));
+
+        return url.toString();
+    };
+};
+
+module.exports = [{
+    name: 'setURLQueryParam',
```

Comment:
I was wondering if we can move this into its own PR. 🍹

Reply:
We could, sure.